Web.R.tex

```diff
@@ -93,16 +93,21 @@ \section{Problems with the current Web} \label{sec:problems}
 This data can be either automatically collected from user interactions, or can be provided directly by the user: consider, as an example, a social network, where users provide personal information such as their generalities, and the website collects data such as post interactions, number and timestamps of logins, and so on.
-This effectively moves ownership of the data from the user to the company. Data that intrinsically belongs to the user (especially personal information such as name, address and phone number) are stored privately into the company server, and the user has limited control over it, since the only possible actions on the data are the ones defined by the company or required by law.
+This effectively moves ownership of the data from the user to the company. Data that intrinsically belongs to the user
+(especially personal information such as name, address and phone number)
+%todo(leandro) - Could be good to use other examples here? I think it could be argued that these are actually kind of public, or expected to be given to a company... some ideas: location data, private messages and documents, search history, photos.
+are stored privately into the company server, and the user has limited control over it, since the only possible actions on the data are the ones defined by the company or required by law.
+%todo(leandro) - What do you think of briefly mentioning/discussing the GDPR?
 
 \section{Thesis goals}
-The goal of this thesis is to present various projects that empower real-case implementations of the distributed Web, as we explain it above. We also aim to analyze notable features and possible defects of each system we present, with particular interest to the aspects that most closely relate to the distributed Web. This thesis is also meant to serve as a reference for seminars, lectures and possibly courses related to the distributed Web.
+The goal of this thesis is to present projects which aim at implementing the distributed Web or solving the problems previously discussed. For each system presented, we analyze its most notable features and discuss possible defects, focusing on aspects most closely related to the distributed Web. This thesis is also meant to serve as a reference for seminars, lectures and possibly courses related to the distributed Web.
 
 The projects described in this thesis have been selected based on the following criteria:
 \begin{itemize}
 	\item A project has as one of its goals to replace or provide an alternative to the existing Web;
 	\item A project must have been successfully deployed and must be still active at the time of writing, with a significant user base;
 	\item A project must mainly depend on data replication in a distributed system and must handle Byzantine failures and adversaries;
+%todo(leandro) - I don't understand what you by "mainly depend on data replication in a distributed system". Would just leaving the second part, about Byzantine failures, be enough?
```

Comment:
The main goal here is to explore the distributed Web, so it makes sense to focus only on services that actually distribute the data and use a distrbuted naming system. You can obtain BFT without relying on distribution.

Reply:
I could argue that you can't get any fault tolerance without distribution (your server crashes it's done).
I don't know, I think just the part about Byzantine failures is enough and less confusing.

Reply:
The reason I wrote that is to exclude projects like Tor and I2P from the Thesis. I would say that these projects are BFT in the sense that communication between clients and hidden services are protected from tampering: we can consider the hidden service server trusted since we trust the content creator and if we assume that he owns the server (which is not always true anymore). But none of these projects rely on replication of data.

While I do explain Tor towards the end of the thesis, it's meant to be just a mention because Tor does not implement a distributed Web. That's why I want to keep it separated from the other projects.

Reply:
I think I understand now.
How about something like:
"A project must tolerate Byzantine failures and provide availability through data replication".